storage: TestReplicaMetrics failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/a65ec243adaa11ea951f02319dc2e02463461ab2

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestReplicaMetrics PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1291012&tab=buildLog

```
=== RUN   TestReplicaMetrics
--- FAIL: TestReplicaMetrics (0.04s)
=== RUN   TestReplicaMetrics/#00
==================
WARNING: DATA RACE
Write at 0x00c00084a518 by goroutine 64:
  github.com/cockroachdb/cockroach/pkg/storage.TestReplicaMetrics.func5()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_test.go:8361 +0x11d
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:827 +0x162

Previous read at 0x00c00084a518 by goroutine 23:
  github.com/cockroachdb/cockroach/pkg/storage.calcReplicaMetrics()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_metrics.go:124 +0x19e
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).Metrics()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_metrics.go:73 +0x63e
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).updateReplicationGauges.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:4150 +0x167
  github.com/cockroachdb/cockroach/pkg/storage.(*storeReplicaVisitor).Visit()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:358 +0x2f1
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).updateReplicationGauges()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:4149 +0x3c8
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).ComputeMetrics()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:4227 +0xd6
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).systemGossipUpdate.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1683 +0x80
  sync.(*Once).Do()
      /usr/local/go/src/sync/once.go:44 +0xde
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).systemGossipUpdate()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1679 +0xd8
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).Start.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1408 +0x95
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:200 +0x145

Goroutine 64 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:878 +0x659
  github.com/cockroachdb/cockroach/pkg/storage.TestReplicaMetrics()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_test.go:8359 +0x4939
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:827 +0x162

Goroutine 23 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:193 +0xc3
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1403 +0x10bb
  github.com/cockroachdb/cockroach/pkg/storage.(*testContext).StartWithStoreConfig()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_test.go:243 +0x159f
  github.com/cockroachdb/cockroach/pkg/storage.TestReplicaMetrics()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_test.go:8167 +0x2bb
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:827 +0x162
==================
    --- FAIL: TestReplicaMetrics/#00 (0.00s)
        testing.go:771: race detected during execution of test

```

SHA: https://github.com/cockroachdb/cockroach/commits/e4806ba5d174c7aeb88ebbefb64d29747803cefa

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestReplicaMetrics PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1292763&tab=buildLog

```
=== RUN   TestReplicaMetrics
--- FAIL: TestReplicaMetrics (0.04s)
=== RUN   TestReplicaMetrics/#00
==================
WARNING: DATA RACE
Write at 0x00c0012ea018 by goroutine 174:
  github.com/cockroachdb/cockroach/pkg/storage.TestReplicaMetrics.func5()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_test.go:8449 +0x11d
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:827 +0x162

Previous read at 0x00c0012ea018 by goroutine 86:
  github.com/cockroachdb/cockroach/pkg/storage.calcReplicaMetrics()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_metrics.go:124 +0x19e
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).Metrics()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_metrics.go:73 +0x63e
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).updateReplicationGauges.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:4150 +0x167
  github.com/cockroachdb/cockroach/pkg/storage.(*storeReplicaVisitor).Visit()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:358 +0x2f1
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).updateReplicationGauges()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:4149 +0x3c8
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).ComputeMetrics()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:4227 +0xd6
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).systemGossipUpdate.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1683 +0x80
  sync.(*Once).Do()
      /usr/local/go/src/sync/once.go:44 +0xde
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).systemGossipUpdate()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1679 +0xd8
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).Start.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1408 +0x95
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:200 +0x145

Goroutine 174 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:878 +0x659
  github.com/cockroachdb/cockroach/pkg/storage.TestReplicaMetrics()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_test.go:8447 +0x4909
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:827 +0x162

Goroutine 86 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:193 +0xc3
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1403 +0x10bb
  github.com/cockroachdb/cockroach/pkg/storage.(*testContext).StartWithStoreConfig()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_test.go:243 +0x1564
  github.com/cockroachdb/cockroach/pkg/storage.TestReplicaMetrics()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_test.go:8255 +0x28b
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:827 +0x162
==================
    --- FAIL: TestReplicaMetrics/#00 (0.00s)
        testing.go:771: race detected during execution of test

```

--- a/pkg/storage/replica_test.go
+++ b/pkg/storage/replica_test.go
@@ -8445,7 +8445,7 @@ func TestReplicaMetrics(t *testing.T) {
 
 	for i, c := range testCases {
 		t.Run("", func(t *testing.T) {
-			zoneConfig := cfg.DefaultZoneConfig
+			zoneConfig := protoutil.Clone(cfg.DefaultZoneConfig).(*config.ZoneConfig)
 			zoneConfig.NumReplicas = proto.Int32(c.replicas)
 
 			// Alternate between quiescent and non-quiescent replicas to test the

